fabric-transfer-api-v1/src/main/java/net/fabricmc/fabric/api/transfer/v1/fluid/base/SingleFluidStorage.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.transfer.v1.fluid.base;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+
+import net.fabricmc.fabric.api.transfer.v1.fluid.FluidPreconditions;
+import net.fabricmc.fabric.api.transfer.v1.storage.Storage;
+import net.fabricmc.fabric.api.transfer.v1.storage.StorageView;
+import net.fabricmc.fabric.api.transfer.v1.transaction.Transaction;
+import net.fabricmc.fabric.api.transfer.v1.transaction.base.SnapshotParticipant;
+
+/**
+ * A storage that can store a single fluid at any given time.
+ * Implementors should at least override {@link #getCapacity}, and probably {@link #markDirty} as well.
+ *
+ * <p>{@link #canInsert} and {@link #canExtract} can be used for more precise control over which fluids may be inserted or
+ * extracted.
+ * If one of these two functions is overridden to always return false, implementors may also wish to override
+ * {@link #supportsInsertion} and/or {@link #supportsExtraction}.
+ */
+public abstract class SingleFluidStorage extends SnapshotParticipant<SingleFluidStorage.State> implements Storage<Fluid>, StorageView<Fluid> {
+	public Fluid fluid;
+	public long amount;
+	private int version = 0;
+
+	/**
+	 * Implement if you want.
+	 */
+	protected void markDirty() {
```

Comment:
Minor detail, but I'd add a short opening sentence about when this is called to the javadoc.